src/test/test-mount-util.c

```diff
@@ -88,6 +88,10 @@ static void test_mnt_id(void) {
                         log_debug("Can't access %s\n", p);
                         continue;
                 }
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to get the mnt id of %s: %m\n", p);
+                        continue;
+                }
```

Comment:
I think it'd be nicer to fold all those three checks into one:
```C
if (r < 0) {
         log_debug_errno(r,
                     r == -EOPNOTSUPP ? "%s doesn't support mount IDs\n" :
                     "Failed to get the mnt id of %s: %m\n",
                     p)
  continue;
}
```

Reply:
Why do you think it would be nicer? I was considering this option, but in the end decided not to do it because I usually prefer not to break crazy `grep`s searching for particular strings that can exist out there.

Reply:
grep would not be impacted, all the strings would still be there. It's nicer because it makes clear that only the log line is different, but otherwise the same handling is applied to all three cases. Also, it's significantly shorter.

Reply:
I'm not sure what strings would still be there because the string "Can't access" is a bit different from the strings "Operation not permitted" and "Permission denied".

Reply:
More generally, I prefer not to break not only crazy `grep`s but all testing tools that somehow handle the output of the tests. It would be very annoying to get a false report from a tool comparing the output of subsequents runs just because something look prettier.

Reply:
I don't think there's any danger of that. Right now the code looks like this
```C
r = ...;
if (r == -EOPNOTSUPP) {
    ...
}
if (IN_SET(r, -EACCES, -EPERM)) {
    ...
}
if (r < 0) {
   ...
}
```
All this verbosity serves no purpose.

Reply:
I'm pretty aware of how the code looks like and I agree that it is far from perfect. Life would probably be easier if the commit where this verbosity was introduced had been reviewed as thoroughly as this one.

Why do you not think that there is any danger of that?

Reply:
@keszybz, I've changed the code, but I decided not to distinguish `EOPNOTSUPP` from any other code, because it doesn't seem to make any sense. I've also pushed one more commit skipping `g_dbus_message_new_from_blob` under `asan`.

Reply:
By the way, this whole thread made me remember my complaining about the breakage of `systemtap` scripts when something had been changed in `libsystemd` :-) I think it is totally fine to make generalizations about what people use and what can and cannot break due to a small change, but it would also be great to somehow clarify why generalizations are made.